lib/rust/ensogl/core/src/display/scene/layer.rs

```diff
@@ -829,6 +881,27 @@ impl std::borrow::Borrow<LayerModel> for Layer {
 
 
 
+// ==================
+// === LayerFlags ===
+// ==================
+
+bitflags::bitflags! {
+    /// A set of flags associated with each [`Layer`].
+    #[derive(Shrinkwrap)]
+    pub struct LayerFlags: u8 {
+        /// When layer is `MAIN_PASS_VISIBLE`, it will be rendered during standard color buffer
+        /// render pass. Layers without this flag are not rendered in main pass, but can still be
+        /// rendered in other passes, e.g. when used as mask layers.
+        const MAIN_PASS_VISIBLE = 1 << 0;
+        /// This layer's camera will be updated every time its parent camera is updated, or the
```

Comment:
Move the LayerFlags definition above layers, so the file can be read from top to bottom, please :)